lua/neogit/lib/git/cli.lua

```diff
@@ -181,6 +181,11 @@ local configurations = {
           return tbl.args(branch)
         end
       end,
+      commit = function(tbl)
```

Comment:
Frankly I find it very difficult when doing these things as I always have to refer to this file and *search* to see if the thing I want to do is implemented, only to find out it sortof is but under a too specific name.

Both `commit` and `branch` just add a normal argument.

It would be far less cognitive load to just `{ "git", "checkout", commit }` as I know what I *want*, and don't necessarily need a semantic wrapper around each of those arguments which has the wrong name given the context.

Again, once `vim.system` lands I consider switching fully to doing raw git commands and wrapping them in higher level concepts like `lib.git.branch.checkout(commit)` upon which you also get lsp completion and help.

Reply:
The current state is just this magic black box where is *sortof* does the right thing, but then inserts things at unexpected places which are different than the order you specified them, and that is without considering the implementation bugs we likely still have in this as it is rather complex.

Not bashing it or anything negative, about it, just that it goes against my first few intuitions I have when approaching programming problems.

Reply:
Thats fair, I've got no special affection for this abstraction. Once the `vim.system` api lands in stable, I'm all for it :)

Reply:
I think the real bummer here is that the libgit2 bindings for lua are written in lua 5.4 😞